# 6장 과제  
컴퓨터공학부  
장수한(2021-14687)


## Problem 1

In [66]:
import numpy as np
from scipy.stats import t
    
# 9명의 점수를 저장하는 리스트 생성
scoreLst = np.array([78, 82, 79, 95, 91, 80, 75, 66, 60])
n = len(scoreLst)

# 평균을 계산
meanVal = scoreLst.mean()
print('표본평균의 값: %.2f' %meanVal)

# 표본표준편차를 계산
stdSVal = np.std(scoreLst, ddof=1)
print('표본표준편차의 값: %.2f' %stdVal)

# 자유도 n-1일 때 t_(alpha/2)의 값을 계산함
alpha = 0.05
tVal = t.ppf(1-alpha/2, n-1)
print('t_0.975의 값: %.2f' %tVal)

표본평균의 값: 78.44
표본표준편차의 값: 10.92
t_0.975의 값: 2.31


In [67]:
lower = meanVal - tVal * stdSVal / np.sqrt(n)
upper = meanVal + tVal * stdSVal / np.sqrt(n)
print('모평균의 95% 신뢰구간: [{:.3f}, {:.3f}]'.format(lower, upper))

모평균의 95% 신뢰구간: [70.049, 86.839]


## Problem 2

전구 수명의 모평균을 $\mu$라고 하자.  

H0: $\mu$ = 190  
H1: $\mu$ > 190  

표본의 크기는 7이다.  
전구의 수명 X ~ N($\mu$, $\sigma^2$)  
$\overline{X}$ ~ N($\mu$, $\sigma^2/7$)  
$\overline{X}-\mu \over S/ \sqrt7$ ~ t(6)

In [68]:
from scipy.stats import ttest_1samp

# 7개 전구의 수명을 저장하는 리스트 생성
durLst = np.array([200, 196, 185, 190, 210, 198, 199])

# ttest_1samp 함수를 이용하여 검정통계량 계산
result = ttest_1samp(durLst, 190)
print('----- ttest_1samp 이용 -----')
print('검정통계량의 값: %.3f \n' %result.statistic)

# 직접 계산하여 검정통계량 계산
num = len(durLst)
durMean = np.mean(durLst)
durStdS = np.std(durLst, ddof=1)
detVal = (durMean - 190) / (durStdS / np.sqrt(num))
print('----- 직접 계산 -----')
print('검정통계량의 값: %.3f\n' %detVal)

isSame = (result.statistic == detVal)
print('두 값의 동일성 여부: {}'.format(isSame))
if isSame: print('두 값은 완벽히 동일하다.')


----- ttest_1samp 이용 -----
검정통계량의 값: 2.289 

----- 직접 계산 -----
검정통계량의 값: 2.289

두 값의 동일성 여부: True
두 값은 완벽히 동일하다.


## Problem 3

H0: $\mu$<sub>A</sub> - $\mu$<sub>B</sub> = 0  
H1: $\mu$<sub>A</sub> - $\mu$<sub>B</sub> > 0

In [69]:
from scipy.stats import ttest_rel

ALst = np.array([4.8, 4.3, 5.4, 4.9, 5.3, 7.7])
BLst = np.array([4.0, 4.2, 5.0, 4.5, 5.6, 7.2])

res = ttest_rel(ALst, BLst)
print('검정통계량의 값이 0보다 큰가?: {}'.format(res.statistic > 0))

# pvalue의 값은 양측검정의 값의 절반이다.
print('p-value: %.6f' %(res.pvalue/2))

검정통계량의 값이 0보다 큰가?: True
p-value: 0.047166


In [70]:
print('-----검정결과-----')
print('p-value 0.047166은 유의수준 0.05보다 작다.')
print('따라서, H0를 기각할 수 있다.\n')
print('유의수준 5%에서 A에 비해 B가 체내 유해물질농도를 줄이는데 더 효과가 있다고 볼 충분한 통계적 근거가 있다.')

-----검정결과-----
p-value 0.047166은 유의수준 0.05보다 작다.
따라서, H0를 기각할 수 있다.

유의수준 5%에서 A에 비해 B가 체내 유해물질농도를 줄이는데 더 효과가 있다고 볼 충분한 통계적 근거가 있다.


## Problem 4

H0: $\sigma$<sub>1</sub><sup>2</sup>/$\sigma$<sub>2</sub><sup>2</sup> = 1  
H1: $\sigma$<sub>1</sub><sup>2</sup>/$\sigma$<sub>2</sub><sup>2</sup> $\neq$ 1

In [71]:
from scipy.stats import f

def var_test(sample1, sample2):
    n1, n2 = len(sample1), len(sample2)
    S1, S2 = np.var(sample1, ddof=1), np.var(sample2, ddof=1)
    dfn, dfd = n1 - 1, n2 - 1

    F = S1 / S2
    pval = 2 * min(f.cdf(F, dfn=dfn, dfd=dfd), 1 - f.cdf(F, dfn=dfn, dfd=dfd))
 
    # 등분산 검정 결과 요약
    print("F test to compare two variances: F = %s, num df = %s, denom df = %s, p-value = %s" 
          % (round(F,5),dfn,dfd,round(pval,5)))

In [72]:
# A 장착 X
noneA = [21, 23, 22, 20, 21, 19, 22, 20, 18]

# A 장착 O
accA = [18, 17, 19, 21, 20, 20, 16]

var_test(noneA, accA)


F test to compare two variances: F = 0.77206, num df = 8, denom df = 6, p-value = 0.71487


유의수준 5%에서 등분산 검정을 실행한 결과  
p-value = 0.71487 > 0.05  
따라서 H0를 기각할 수 없다.  

즉 등분산을 가정할 충분한 통계적 근거가 있다.  

## Problem 5

H0: $\mu$<sub>A미장착</sub> - $\mu$<sub>A장착</sub> = 0  
H1: $\mu$<sub>A미장착</sub> - $\mu$<sub>A장착</sub> > 0

In [73]:
from scipy.stats import ttest_ind

result = ttest_ind(noneA, accA, equal_var=True)
print(result)

Ttest_indResult(statistic=2.3085184093649356, pvalue=0.03675156998512593)


In [74]:
print('검정통계량이 양수인가?: {}'.format(result.statistic > 0))
# 단측검정이므로 p-value의 값을 양측검정의 절반으로 한다.
print('p-value: %.6f' %(result.pvalue/2))

검정통계량이 양수인가?: True
p-value: 0.018376


유의수준 5%에서는 p-value의 값 0.018376이 0.05보다 작으므로 H0를 기각할 수 있다.  
즉, 유의수준 5%에서 A의 장착이 연비저하를 일으킨다고 볼 충분한 통계적 근거가 있다.